### PR TITLE
feat: fetch compiled releases from an Ops Manager bosh director

### PR DESCRIPTION
**Pull request by crhntr**
Todo
- [x] Calculate list of non-compiled releases (aka built releases) from Kilnfile.lock
- [x] Use OM variables to auth with ops manager
- [x] Proxy through the Ops Manager to the Bosh Director (and authorize with creds from OM)
- [x] Download and checksum the releases
- [x] Upload the releases to an s3 release source
- [x] Update the Kilnfile.lock
- [x] Fix accidental renames in help/acceptance packages
- [x] Consider testing the networking/proxy code in the om package
- [ ] Manually test
- [ ] Code review

### CODE REVIEW

**Comment by crhntr** [comment]
Blocked-By: https://github.com/pivotal-cf/kiln/pull/264

**Comment by crhntr** [comment]
Decided against "...testing the networking/proxy code in the om package." I'll do it in the future. Right now this seems to work.

**Comment by crhntr** [comment]
I rebases this with main. Instead of fixing the mod stuff in place I accepted theirs and then re-added the om dependency as a final commit. So intermediate commits on this branch might be broken.

File: internal/acceptance/help_test.go [hunk 1]
```diff
@@ -17,20 +17,21 @@ Usage: kiln [options] <command> [<args>]
...
-  pre-process             preprocess yaml files
-  publish                 publish tile on Pivnet
-  sync-with-local         update the Kilnfile.lock based on local releases
-  update-release          bumps a release to a new version
-  update-stemcell         updates stemcell and release information in Kilnfile.lock
-  upload-release          uploads a BOSH release to an s3 release_source
-  validate                validate Kilnfile and Kilnfile.lock
-  version                 prints the kiln release version
+  bake                     bakes a tile
+  cache-compiled-releases  Cache compiled releases
```
Comment (ciriarte) [inline internal/acceptance/help_test.go:21]: **nitpick:** this is the only help item starting with a capitalized letter.

File: internal/commands/cache_compiled_releases.go [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+)
+
+type CacheCompiledReleases struct {
+	Options struct {
+		flags.Standard
+		om.ClientConfiguration
+
+		UploadTargetID string `           long:"upload-target-id"   required:"true"    description:"the ID of the release source where the built release will be uploaded"`
+		ReleasesDir    string `short:"rd" long:"releases-directory" default:"releases" description:"path to a directory to download releases into"`
+		Name           string `short:"n"  long:"name"               default:"cf"       description:"name of the tile"` // TODO: parse from base.yml
```
Comment (ciriarte) [inline internal/commands/cache_compiled_releases.go:48]: **question:** There's a TODO here. Do we still need to do this?

**Review by ciriarte — APPROVED** [review]
**thought:** 🧐 Smug face with security TLS skipping, but we will merge it :)

Thank you so much @crhntr this is awesome.

**Review by ciriarte — APPROVED** [review]
**thought:** 🧐 Smug face with security TLS skipping, but we will merge it :)

Thank you so much @crhntr this is awesome.

**Review by ciriarte — APPROVED** [review]
**thought:** 🧐 Smug face with security TLS skipping, but we will merge it :)

Thank you so much @crhntr this is awesome.

**Review by ciriarte — APPROVED** [review]
**thought:** 🧐 Smug face with security TLS skipping, but we will merge it :)

Thank you so much @crhntr this is awesome.

**Pull request merged**